### PR TITLE
Add MurmurHash3 Operation

### LINKED ISSUE #1375: Feature request: Support Murmur-Hash
In short ... it would be great if CyberChef supported Murmur-Hash which is also used in e.g. shodan.io .

### PR DESCRIPTION
**Pull request by AliceGrey**
Fixes #1375

### CODE REVIEW

**Comment by AliceGrey** [comment]
This is useful for generating favicon hashes for threat hunting in Shodan
Recipe:
```
To_Base64('A-Za-z0-9+/=')
Find_/_Replace({'option':'Regex','string':'(.{76})'},'$1\\n',true,true,true,true)
Find_/_Replace({'option':'Regex','string':'$'},'\\n',true,false,false,false)
MurmurHash3(0)
```

File: src/core/operations/Murmurhash3.mjs [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/**
```
Comment (a3957273) [inline src/core/operations/Murmurhash3.mjs:1]: This file seems to verbatim copy the MIT Murmur implementation by GaryCourt, so should likely include their license information.  See an example here: https://github.com/gchq/CyberChef/blob/df151eabf91cdb31292cfa336844bfdc24afba19/src/core/operations/ToQuotedPrintable.mjs#L4

File: src/core/operations/Murmurhash3.mjs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        let seed;
+        if (args && args.length > 0) {
+            seed = args[0];
+        }
+        return this.mmh3(input, seed);
+    }
+
+}
+
+export default MurmurHash3;
```
Comment (a3957273) [inline src/core/operations/Murmurhash3.mjs:113]: Could you add some tests, so we know this implementation is correct?

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: src/core/operations/Murmurhash3.mjs (hunk 2 above)
Comment (AliceGrey) [reply src/core/operations/Murmurhash3.mjs:117]: Added a few tests 👍

**Comment by AliceGrey** [comment]
Rebased

**Comment by AliceGrey** [comment]
I believe I may have found a bug. Converting it back to a draft for now.

**Comment by AliceGrey** [comment]
Should be g2g now

**Comment by a3957273** [comment]
Thanks for the contribution @AliceGrey, this looks really good. :)

**Pull request merged**